aerogear/mobile-security-service
# fix(ui): no rerender on text input and save local state

## Motivation
https://issues.jboss.org/browse/AEROGEAR-8698

## What
Store the state of the rows locally once text is entered in the input field or a checkbox is changed
Added version object to make the manipulation of the rows easier.

## Verification Steps
 
1. Run the application in development mode with a populated database
2. Using the Chrome React Plugin, navigate down to AppVersionsTableContainer:
Expand State:

_Expected:_

3. After initial page render: shouldAppUpdate = false
4. After initial page render: updatedVersions exist with an array of row data which matches appVersions: state.app.versionsRows (this can be removed later when we remove rows from state.
5. Make a change to any of the Input boxes, this should be reflected in the updatedVersions object
6. Make a change to any of the checkboxes, this should be reflected in the updatedVersions object

7. The screen should not rerender on each local state change and as such the text in the text box is visible and the text box doesn't lose focus.


## Checklist:


## Additional Notes

@craicoverflow I thought that there wasn't enough progress in this to make it a seperate pr but maybe take a look and see if we could use this as is or with some changes. Let's chat about it in the morning.
